### ISSUE TITLE
How to write a loss function and metric function by self-define in python? use cntk.ops?

### ISSUE THREAD

**Issue by whaozl** [Microsoft/CNTK#1681]
cntk2.0-beta15 provides

some loss, for example cross_entropy_with_softmax、weighted_binary_cross_entropy、cosine_distance、binary_cross_entropy、squared_error,

som metric function for example classification_error、edit_distance_error、ndcg_at_1、lambda_rank.

**but who can help me? I want to self define a loss function and metric fucntion for example IOU in Picture segement.**

**Comment by cha-zhang** [comment]
Follow example here: https://github.com/Microsoft/FERPlus

**Comment by whaozl** [comment]
Thank you very much, I see in your link[https://github.com/Microsoft/FERPlus](https://github.com/Microsoft/FERPlus). 

```
# Cross Entropy.
train_loss = ct.negate(ct.reduce_sum(ct.element_times(target, ct.log(prediction)), axis=-1))

pe         = ct.classification_error(z, label_var)
```

but How to write a metric function by self-define in python? Thank you.
